agent/consul/replication.go

```diff
@@ -104,7 +104,7 @@ func (r *Replicator) Run(ctx context.Context) error {
 			// the next round of replication
 			atomic.StoreUint64(&r.lastRemoteIndex, 0)
 
-			if r.suppressErrorLog != nil && !r.suppressErrorLog(err) {
+			if r.suppressErrorLog == nil || (r.suppressErrorLog != nil && !r.suppressErrorLog(err)) {
```

Comment:
I think this could be also written this way:
```suggestion
			if r.suppressErrorLog == nil || !r.suppressErrorLog(err) {
```

because `(r.suppressErrorLog != nil && !r.suppressErrorLog(err))` is only checked if `r.suppressErrorLog != nil`.

Reply:
Ah yep, you are right, thanks! Finally got to this after the holiday break. :)